### ISSUE TITLE
Citation/bibcit problems with Orlando schema/stylesheet (embedded CWRC-Writer)

### ISSUE THREAD

**Issue by SusanBrown** [cwrc/CWRC-Writer#454]
I am about to flag a number of problems with citations. 

I suspect these problems are a mix of stylesheet problems and problems with the editor. I'm flagging them to the person I think should first look at them.

@jefferya but perhaps with some advice from @ajmacdonald on stylesheets/using icons or popups
- [ ] An existing bibcit formats with the icon from the Orlando styles sheets and the contents of the placeholder attribute in parentheses. This is logical in one sense but really what should be rendered in parentheses is text of the tag, with the placeholder information available somehow e.g. through a popup
- [ ] once we are using citation entities/URIs instead of DBREFS, we would want to provide some kind of short form information for that text and a link to the full record.

**Comment by jefferya** [comment]
@SusanBrown 

There is little I can do with CSS as it is not really meant for styling XML beyond a simplistic manner (i.e., JavaScript or XSLT are needed for more heavy lifting or changes to the underlying documents ). The following is what I've been able to accomplish in 2.5+ hours - https://github.com/cwrc/CWRC-Schema/commit/2a615cc4e5cd114c4e6d5673ab645cb48db746cf.

In my opinion, this approach is severely limited and the previous might be better though one of the other reported issues might now work better: https://github.com/cwrc/CWRC-Writer/issues/455
The problems with the above CSS only approach include:
* CSS pseudo-elements (:before/:after) required to add parenthesis around the text of the BIBCIT element but these pseudo-elements overridden by the CSS needed for the hover popup containing the PLACEHOLDER value so the parenthesis disappear whenever the hover popup is present. This is a limitation of the CSS approach 
* If the text of the BIBCIT is very short and occurs at the end of the line near the right margin then the hover popup will be cutoff by the scrollbar. Maybe there is a way to prevent this but not one I could find. Another option, have the PLACEHOLDER appear in-line with the text but then the existing text is being pushed around. 
* Disorienting to edit the BIBCIT because Javascript used to build the right-click popup but things are moving around due to the CSS hover on a psuedo element displaying the PLACEHOLDER value.  

I may be misunderstanding CSS so if there is a better way, let me know.

The best approach would be to ditch the CSS and use a JavaScript approach to build the popups in my opinion.

**Comment by SusanBrown** [comment]
@jefferya @ajmacdonald 

Thanks for your efforts, Jeff. Sounds very frustrating. I've been thinking that perhaps what might make most sense is to have a meeting of the three of us to talk through a set of problem to do with editing and formatting the bibcits and scholarnotes since they extend beyond Orlando. 

Would you both have time later this afternoon/early evening or mid-late-afternoon (EDT) tomorrow?

**Comment by SusanBrown** [comment]
p.s. @jefferya can I see what you've done to the CSS in action within beta?

**Comment by jefferya** [comment]
@SusanBrown Yes, visible within beta and also the standalone, cwrc-dev-05, and Echidna sandbox (cwrc-dev-06)

I should be available the around the time you mentioned.

**Comment by ajmacdonald** [comment]
@SusanBrown I'll be available tomorrow afternoon

**Comment by ajmacdonald** [comment]
A bibcit popup could use code similar to that already in place for link popups ( https://github.com/cwrc/CWRC-Writer/issues/369 )

**Comment by ajmacdonald** [comment]
Done: https://github.com/cwrc/CWRC-Writer/commit/babf78f1a850f0b3ed59232f34f1cc6e80935a78

**Comment by SusanBrown** [comment]
Fabulous. What's the next step @jefferya ?

**Comment by jefferya** [comment]
@SusanBrown 

Next step, change the islandora_cwrc_writer module.

The details:

Tested in the standalone CWRC-Writer with the doc [1] and it looks ok

Tested with the following object in the embedded CWRC-Writer and it failed. Likely part of the following module, islandora_cwrc_writer, needs to be changed to reflect the latest CWRC-Writer changes  https://github.com/jefferya/islandora_cwrc_writer/tree/7.x/js 

[1]
http://apps.testing.cwrc.ca/editor/dev/editor_dev.htm#zzzz

[2]
http://cwrc-dev-05.srv.ualberta.ca/islandora/object/cwrc%3A8d235859-a651-435b-b152-f034ba1a22c7/edit-cwrc

**Comment by jefferya** [comment]
@SusanBrown 

The popups are invisible in the embedded versions of CWRC-Writer (or rendering in some weird location). The popup.js functions seem to be correctly called in the embedded version but one can't see the popup.

Likely need @ajmacdonald's help.

**Comment by ajmacdonald** [comment]
Hopefully it's now fixed and hasn't broken other popups. Please test.

**Comment by SusanBrown** [comment]
@colinraybrian Again, this is fixed, I think. Pls test.
